### PR TITLE
mempool & rpc log grooming

### PR DESCRIPTION
**Pull request by alexanderbez**
I have this set to be backported for 0.34.x. Obviously the small shim change will cause a conflict, but that'll be very easy to address.

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]

File: rpc/jsonrpc/server/http_json_handler.go
Comment (alexanderbez) [reply rpc/jsonrpc/server/http_json_handler.go:101]: SAVAGE 😆

File: test/e2e/runner/rpc.go
Comment (cmwaters) [inline test/e2e/runner/rpc.go:102]: I was going to implement something that depended on the size of the network i.e. more nodes = longer wait

**Review by cmwaters — APPROVED** [review]

File: test/e2e/runner/rpc.go
Comment (alexanderbez) [reply test/e2e/runner/rpc.go:102]: Yeah, this should be configurable somewhat. Let's hold off on merging my PR until yours lands 👍

**Pull request merged**
